backend/molgenis-emx2-rdf/src/main/java/org/molgenis/emx2/rdf/RDFService.java

```diff
@@ -470,23 +486,31 @@ private String getLabelForRow(final Row row, final TableMetadata metadata) {
     return String.join(" ", primaryKeyValues);
   }
 
-  private List<Row> getRows(final Table table, final String rowId) {
+  private List<Row> getRows(Table table, final String rowId) {
     Query query = table.query();
     if (rowId != null) {
+      // first find from root table
       PrimaryKey key = PrimaryKey.makePrimaryKeyFromEncodedKey(rowId);
-      query.where(key.getFilter());
-    }
-    // If a table is extended then we get only those rows that are for the base table.
-    if (table.getMetadata().getColumnNames().contains("mg_tableclass")) {
-      var tableName = table.getSchema().getName() + "." + table.getName();
-      query.where(f("mg_tableclass", EQUALS, tableName));
+      List<Row> oneRow = query.where(key.getFilter()).retrieveRows();
+      // if subclass
```

Comment:
Explain why this is needed for a subclass

Reply:
renamed function and added some docs hope that helps